specs/number-field.md

```diff
@@ -0,0 +1,123 @@
+# Number Field
+
+## Overview
+
+An implementation of a generic input for numerical values as a form-connected web-component.
+
+### Use Cases
+
+Used anywhere an author might otherwise use input[type="number"].
+
+### Features
+
+1. Form association
+
+2. Conversion to/from number/string between the view and the control
+
+### Prior Art/Examples
+
+- [FAST-DNA (React)](https://explore.fast.design/components/number-field)
+- [Carbon Design](https://www.carbondesignsystem.com/components/number-input/code)
+- [Ant Design](https://ant.design/components/input-number/)
+- [Office Fabric](https://developer.microsoft.com/en-us/fabric#/controls/web/spinbutton)
+- [Duet](https://www.duetds.com/components/number-input/)
+
+---
+
+### API
+
+Extends Form associated component
+
+*Component Name*
+- `fast-number-field`
+
+*Attrs* - Adapted from [MDN](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/number)
+- `autofocus` - automatically focuses the control
+- `readonly` - the number field should be submitted with the form but should not be editable.
+- `placeholder` - an exemplar value to display in the input field whenever it is empty
+- `required` - boolean value that sets the field as required
+- `value` - string value of the number field, can be an empty string
+- `disabled` - disables the control
+- `list` - the id of the `<datalist>` element that contains the optional pre-defined autocomplete options
+- `max`	- the maximum value to accept for this input
+- `min` -	the minimum number to accept for this input
+- `maxlength`	- the maximum number of characters the input should accept
+- `minlength` -	the minimum number of characters long the input can be and still be considered valid
+- `name` - the name of the control
+- `size` - a number indicating how many characters wide the input field should be
+- `appearance` - enum
+  - outline (default)
+  - filled
+
+*Events*
+- `change: CustomEvent`
+  - no custom data
+  - bubbles
+
+### Anatomy and Appearance
+
+```HTML
+<!-- shadow root -->
+<label part="label"><slot></slot></label>
+<div part="root">
+  <slot name="start" part="start"></slot>
+  <input type="number" part="control" />
+  <slot name="end" part="end"></slot>
+</div>
+<!-- end shadow root -->
+```
+
+
+*Screenshots below are of the basic appearance of the component and are not exhaustive.*
+
+**The visual treatment of this control should visually match the number field component**
+
+| State | Image |
+| ----- | ----- |
+| default | ![](../text-field/images/text-field.png) |
+| focus | ![](../text-field/images/text-field-focus.png)
+| filled | ![](../text-field/images/text-field-filled.png)
+
+*Slot Names*
+- default - the label content
+- start - often times a glyph, icon, or button precedes input
+- end - often times a glyph, icon, or button follows the input
+
+*Host Classes*
+- disabled
+- required
+- readonly
+
+*CSS Parts*
+- label
+- root
+- start
+- end
+- control
```

Comment:
Would we provide custom spin buttons with the ability to style them and replace their icon?

Reply:
I think we'll provide slots for it, but I think the buttons themselves might be a different spec and PR.

Reply:
Initially I think the spin buttons could be passed into the existing `end` slot. I'll add next steps for creating those and passing them by default.